libretro-common/lists/string_list.c

```diff
@@ -228,6 +235,9 @@ bool string_list_append_n(struct string_list *list, const char *elem,
 void string_list_set(struct string_list *list,
       unsigned idx, const char *str)
 {
+   if (!list || !str || (idx >= list->size))
```

Comment:
If !list or idx >= list->size, that is a bug in the caller, and we should crash or log a complaint (assert(), for example) so the caller's bug can be fixed.

This kind of try { ... } catch { /* nothing */ } equivalents exist just about everywhere in the codebase; please don't introduce even more.

And if !str, we should either complain or set the entry to empty, not just leave it unchanged.